entity-framework/core/querying/database-functions.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Database Functions - EF Core
+description: Information on database functions supported in EF Core query translation
+author: smitpatel
+ms.date: 11/10/2020
+uid: core/querying/database-functions
+---
+# Database Functions
+
+Database functions are database equivalent of [C# methods](/dotnet/csharp/programming-guide/classes-and-structs/methods). A database function can be invoked with zero or more parameters and it computes the result based on the parameter values. Most databases, which use SQL for query have support for database functions. So SQL generated by EF Core query translation also allows invoking database functions. C# methods don't have to translate to database functions strictly in EF Core.
+
+- A C# method may not have an equivalent database function.
+  - <xref:System.String.IsNullOrEmpty%2A?displayProperty=nameWithType> method translates to a null check and a comparison with an empty string in the database rather than a function.
+  - <xref:System.String.Equals(System.String,System.StringComparison)?displayProperty=nameWithType> method doesn't have database equivalent since string comparison can't be represented or mimicked easily in a database.
+- A database function may not have an equivalent C# method. `??` operator in C#, which doesn't have any method, translates to `COALESCE` function in database.
+
+## Types of database functions
+
+EF Core SQL generation supports a subset of functions that can be used in database. This limitation comes from the ability to represent query in LINQ for the given database function. Further, each database has varying support of database functions, so EF Core provides a common subset. A database provider is free to extend EF Core SQL generation to support more patterns. Following are type of the database functions EF Core supports and uniquely identifies. These terms also help in understanding what translations come built in with EF Core provider.
+
+### Built-in vs user-defined functions
+
+Built-in functions come with database predefined, but user-defined functions are explicitly defined by the user in the database. When EF Core translates query to use database functions, it uses built-in functions to make sure that the function is always available on database. The distinction of built-in function is necessary in some databases to generate SQL correctly. For example SqlServer requires that every user-defined function is invoked with schema qualified name. But built-in functions in SqlServer don't have any schema. PostgreSQL defines built-in function in `public` schema and it can be invoked with schema qualified name.
+
+### Aggregate vs scalar vs table-valued functions
+
+- Scalar functions take scalar values as parameter(s) and return a scalar value as the result. Scalar functions can be used anywhere in SQL where a scalar value can be passed.
```

Comment:
Maybe add an example of a *scalar value* for clarity.

Reply:
Like add ?
 Scalar values are the values of primitive types supported by the database.
What kind of examples should I add? actual values - 1, "Seattle"
or server types like int, text, nvarchar(max)
or client types like int,string ?

Reply:
Those are good. Even something more subtle would be fine:

...take scalar values—like integers or strings—as parameters...